ecl/resource_ecl_provider_connectivity_tenant_connection_v2.go

```diff
@@ -0,0 +1,461 @@
+package ecl
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+
+	"github.com/nttcom/eclcloud"
+	"github.com/nttcom/eclcloud/ecl/provider_connectivity/v2/tenant_connections"
+)
+
+func resourceProviderConnectivityTenantConnectionV2() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceProviderConnectivityTenantConnectionV2Create,
+		Read:   resourceProviderConnectivityTenantConnectionV2Read,
+		Update: resourceProviderConnectivityTenantConnectionV2Update,
+		Delete: resourceProviderConnectivityTenantConnectionV2Delete,
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"tags": &schema.Schema{
+				Type:     schema.TypeMap,
+				Optional: true,
+			},
+			"tenant_connection_request_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"device_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"ECL::Compute::Server",
+					"ECL::Baremetal::Server",
+					"ECL::VirtualNetworkAppliance::VSRX",
+				}, false),
+			},
+			"device_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"device_interface_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"attachment_opts_server": &schema.Schema{
```

Comment:
How about separating this to bare metal and compute server?

Reply:
It might be good. Consider.